### PR TITLE
Set up maintainer workflow

### PR DESCRIPTION
**Pull request by JustinLivi**
@fahad19 Checking to make sure I set this up properly and that we're going to increment minor semver levels as opposed to releasing 1.0.0

The automatically generated Change log should be empty at this point, correct?

If everything looks good, I can run `make release VERSION=minor`

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+release:
+	echo "Releasing version: $(VERSION)"
+	git checkout master
+	git pull origin master
+	npm run transpile
+	npm run dist
+	npm version $(VERSION)
+	npm publish
+	git push --follow-tags
+	npm run docs:publish
```
Comment (fahad19) [inline Makefile]: we don't need all the commands, for example we don't use `gitbook`, so this line could be removed.

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# Change Log
```
Comment (fahad19) [inline CHANGELOG.md:1]: I was lazy before, and didn't create any `tags`, so it is empty for now. and that is fine 👍

File: Makefile (hunk 1 above)
Comment (fahad19) [reply Makefile]: this set of commands should do:

```
echo "Releasing version: $(VERSION)"
git checkout master
git pull origin master
npm version $(VERSION)
npm publish
git push --follow-tags
```

**Review by fahad19 — APPROVED** [review]
left some comments. really good PR! 👍

**Comment by JustinLivi** [comment]
Oops. Hadn't noticed the extra release commands in there. Good catch! Removed those so I think this looks good to release. If I get some free time, I'll try and add some tests.

**Comment by fahad19** [comment]
💖

**Comment by fahad19** [comment]
@JustinLivi: can you please verify if the last command in `make release` worked?

It is `git push --follow-tags`, which should have created a tag here: https://github.com/fahad19/winston-slack-hook/tags

Changing it to `git push origin master --follow-tags` might help too, if the previous one didn't work for you.

**Comment by JustinLivi** [comment]
@fahad19 You're right it hadn't created them on the master repository, just my own fork. I have the master set up as `upstream` locally so for me I needed to run `git push upstream master --follow-tags`. We can leave the release script using `git push --follow-tags` if that works for you. It's simple enough for me to run the additional command. Alternatively, we could try to find a more general solution.

**Comment by fahad19** [comment]
cool with me. thanks @JustinLivi!

**Pull request merged**